src/containerapp/azext_containerapp/java_component_decorator.py

```diff
@@ -214,3 +216,126 @@ def process_loaded_scg_route(self):
+        else:
+            self.java_component_def["properties"]["configurations"] = current_java_component_def["properties"]["configurations"]
+
+    def update(self):
+        try:
+            return self.client.update(
+                cmd=self.cmd, resource_group_name=self.get_argument_resource_group_name(),
+                environment_name=self.get_argument_environment_name(), name=self.get_argument_java_component_name(),
+                java_component_envelope=self.java_component_def,
+                no_wait=self.get_argument_no_wait())
+        except Exception as e:
+            handle_raw_exception(e)
+
+    def add_or_update_configurations(self, existing_configurations, new_configurations):
+        for new_configuration in new_configurations:
+
+            # Check if updating existing configuration
+            is_existing = False
+            for existing_configuration in existing_configurations:
+                if existing_configuration["propertyName"].lower() == new_configuration["propertyName"].lower():
+                    is_existing = True
+
+                    if "value" in new_configuration:
+                        existing_configuration["value"] = new_configuration["value"]
+                    else:
+                        raise ValidationError("Value must be provided for configuration {}.".format(new_configuration["propertyName"]))
+
+            # If not updating existing configuration, add it as a new configuration
+            if not is_existing:
+                existing_configurations.append(new_configuration)
+
+    def remove_configurations(self, existing_configurations, remove_configurations):
+        for remove_configuration in remove_configurations:
+
+            # Check if updating existing configuration
+            is_existing = False
+            for i, value in enumerate(existing_configurations):
+                existing_configuration = value
+                if existing_configuration["propertyName"].lower() == remove_configuration.lower():
+                    is_existing = True
+                    existing_configurations.pop(i)
+                    break
+
+            # If not updating existing configuration, add it as a new configuration
+            if not is_existing:
+                logger.warning("configuration {} does not exist.".format(remove_configuration))
+
+    def set_up_set_configurations(self):
+        if self.get_argument_set_configurations() is not None:
+            configuration_list = self.parse_configurations(self.get_argument_set_configurations())
+            self.add_or_update_configurations(self.java_component_def["properties"]["configurations"], configuration_list)
+
+    def set_up_replace_configurations(self):
+        if self.get_argument_replace_configurations() is not None:
+            configuration_list = self.parse_configurations(self.get_argument_replace_configurations())
+            self.java_component_def["properties"]["configurations"] = []
```

Comment:
just set it to `configuration_list` and skip the next method call?

Reply:
Cool, that's better, updated